### ISSUE TITLE
Image file disappears after modification when created from template, displaying an endless «Loading...»

### ISSUE THREAD

**Issue by SebastJava** [linuxmint/nemo#2736]
```
 * Nemo version: 4.8.6
 * Issue with windowed nemo
 * Distribution: Linux Mint 20.1 Ulyssa
 * Graphics:  Device-1: Intel 2nd Generation Core Processor Family Integrated Graphics driver: i915 v: kernel 
           Display: x11 server: X.Org 1.20.9 driver: modesetting unloaded: fbdev,vesa resolution: 1920x1080~60Hz 
           OpenGL: renderer: Mesa DRI Intel HD Graphics 2000 (SNB GT1) v: 3.3 Mesa 20.2.6
 * 64 bit
 ```
![ImgDisappeared](https://user-images.githubusercontent.com/31940713/115434683-f7f47b00-a1d6-11eb-8b84-1c8c51caa655.png)

**Issue**
Image file disappears after modification if image was created from template

**Steps to reproduce**
* Have **thumbnails enabled**
* Create any SVG or PNG file, with any software, and save it into **~/Templates/** directory
* Create a **~/Test/ directory**
* Right-click in your ~/Test/ directory to **Create New Document > file-from-template**
* **Change directory to parent**, so ~/Test/file-from-template is not visible but without closing Nemo.
* Open ~/Test/file-from-template from any software, **do some change, and save it**. (1)
* **Open ~/Test/ directory**: file-from-template disappeared, displaying an endless «Loading...»

**Expected behaviour**
I would like to be able to edit images without worrying about them disappearing.

**Other information**
Sorry to bother you again. There are lots of similar issues with thumbnails.
* I am trying to narrow down the problem, so, **all my plugins were disabled**.
* I also had similar issues with thumbnails on Caja, on Mint MATE 18.3 32-bit.
________________________
1.  You could also have opened this file-from-template from Nemo itself, but before you save and before you move up to parent directory, of course.

**Comment by SebastJava** [comment]
**Other information (bis)**
* View > Show Thumbnails: on/off is overridden by Edit > Preferences > Preview... Show thumbnails
* I have a feeling that sometimes creating the thumbnails takes double the time. But this is just a feeling

**Comment by leigh123linux** [comment]
I'm going to  unsubscribe due to your comment spam.

**Comment by SebastJava** [comment]
@mtwebster Did you see this? Sorry to bother you again but you worked on a similar issue not long ago.

**Comment by Jeremy7701** [comment]
But note that Edit > Preferences > Preview... also includes an option of "Inherit thumbnail visibility from parent".
Once nemo has created a thumbnail, it tends to keep it - switching off thumbnailing supplies a default icon.

Personally, since I mostly use the icon view, I have **all** thumbnail preferences set as permanently 'yes' - including having the toolbar icon that allows switching on thumbnails on/off permanently set to 'on'.

**Comment by SebastJava** [comment]
![image](https://user-images.githubusercontent.com/31940713/118377643-81138d80-b59c-11eb-8cd6-062329ded1f6.png)

@Jeremy7701 These are my current settings in Nemo preferences. And i removed the toolbar icon for thumbnails on/off because i never toggle this off. Re-checked in Nemo > View > Show Thumbnails: ON.

I don't think i changed these settings lately, but, just in case, i will re-test these "steps to reproduce"... RE-TESTED! I still get this same bug. And ALL the Plugins > Actions and Extensions are currently Disabled.

**Comment by SebastJava** [comment]
@Jeremy7701 Can you reproduce this issue?

**Comment by Jeremy7701** [comment]
I can't reproduce your fault - the "view" option takes precedence over the "preferences" option, as it should. I've have most off the addons loaded. My settings are

![Screenshot from 2021-05-16 12-36-47](https://user-images.githubusercontent.com/52585119/118395743-e53d5c80-b643-11eb-80bd-ff4f37fdd6d5.png)

BTW I've got a new PC, so I've got a completely fresh install of LMDE4, only about 4 months old. Have you tried loading Mint/LMDE as a VM (so that you get a completely fresh install)?
Have to say, I think this kind of bug would have been spotted before now, if it were widespread.

**Comment by SebastJava** [comment]
@Jeremy7701 I still have to look at your exact preferences, and think about it, and try it. But there's 1 or 2 things i want to give a quick response to...
1. I experienced those similar issues on a fresh LM Cinnamon 20 install. At the time we did all those tests, it was still approximately fresh. But, okay, i did some "hazardous experiments" during this #2623 investigation. VM? I never tried that, don't know how to, and it would probably be slow on my old Dell Optiplex 790 (i5). But, hey, i still have this LM 20 USB key installer. That's like totally new! I will test on this, soon.
2. Not widespread!? I think it is widespread. And old. I also experienced this similar issue #2623, long ago, on my netbook with LM MATE 18.3. That's with the Caja file manager. And... Because it recently reached EoL, i just did a complete fresh install of LM MATE 19.3, just a few days ago, on this little netbook. I will do the test on this too.

Oh, i see you were just talking about this

> "view" option takes precedence over the "preferences" option

but did you get **this issue,** as described in the Steps to reproduce, on top of this page? Remember, please follow exactly each steps, including this **key step:**

> Change directory to parent, so ~/Test/file-from-template is not visible but without closing Nemo.

I still have to do those double and triple checks on my USB bootable key with LM Cinnamon 20 and also this fresh LM MATE 19.3. But i am pretty sure i will get this same bug. Stay tuned!

P.S.: If you really can't reproduce this issue, then i may have to narrow it down to what software to use, what kind of image, etc.
P.S.: You are on LMDE4. If you have a lot of spare time, it would be appreciated if you could try this on the exact same system as me: LinuxMint Cinnamon 20.1 64-bit...

**Comment by Jeremy7701** [comment]
I didn't appreciate the significance of the **template** and then opening the file outside nemo.
So yes - I can reproduce the original issue, I'm not sure that the view option is relevant, though.

Furthermore, most templates such as writer/Calc/text don't have a file-dependant thumbnails - actually, if I was creating an image or video, I wouldn't find the template option to be especially useful as it only provides a single file per user.
Why not create a single "template-file" in your working directory and use edit + save-as to create multiple copies?

This works, provided you launch inkscape from nemo; all the different thumbnails are created.

![Screenshot from 2021-05-16 17-52-22](https://user-images.githubusercontent.com/52585119/118405367-88579b80-b66f-11eb-8c49-1fe3e15e6cd9.png)

**Comment by SebastJava** [comment]
@Jeremy7701

> I didn't appreciate the significance of the **template** and then opening the file outside Nemo.

You could also have opened this file-from-template from Nemo itself, but before you save and before you move up to parent directory, of course. See my edited **Steps to reproduce** here on top. ▲

> So yes - I can reproduce the original issue

Ah. So i am not crazy. Guess I'm exempt from having to install a brand new system? I could also have restarted on this USB key, but that's much less relevant because that way I wouldn't have the very latest version of Nemo.

> I wouldn't find the template option to be especially useful as it only provides a single file per user.

I can very well imagine several situations where the templates+thumbnails would be very relevant.

**Example:** I create a newsletter in Inkscape. That makes 8 to 16 pages, and it is published every month. One of the basic rules for good design is repetition: pages 2 to 16 must share some common elements such as a unique logo, a unique page numbering, plus some top and bottom page lines and layouts, etc. So, it would really make sense to start creating every page from a common template. And that's 16 page X 12 months = 192 pages created neatly and easily.

And on top of all that, i am not sure this bug only occurs with Templates... But I avoid creating new issues unless i can give precise instructions on how to consistently reproduce the issue, every time, after having it tested and retested.

Hey, this looks great. I am writing this long post here, and this JavaScript or whatever state-of-the-art dynamic web page got this red symbol added along with this text: "**mtwebster** closed this in `73aafd8` 22 minutes ago". So, that's it. Hopefully. I can't wait to see this new Nemo version... And test it again. :)
